### ISSUE TITLE
Fault exception, only occurs on iOS

### ISSUE THREAD

**Issue by mlaframboise** [alebianco/ANE-Google-Analytics#39]
I am currently logging events using the following line:
m_googleTracker.buildEvent(category, action).withLabel(label).track();

Everything works correctly on Android, but whenever I call this line on iOS, I get the following error:
[Fault] exception, information=ArgumentError: Unable to read a property on method 'trackTiming'.

To my knowledge, it shouldn't even be calling trackTiming since I'm using the buildEvent function, not buildTiming. Is something broken in the iOS code?

**Comment by alebianco** [comment]
There's no way that that error is triggered by your as3 code. you must be tracking a timer somewhere ...

**Comment by mlaframboise** [comment]
Nope, I've checked this a number of times. This is literally the only call in the entire application.

When I build for Android, this call works as expected, but once I build for iOS, it crashes with the posted error every time. There's something in the iOS code that is calling trackTiming when it should be using trackEvent instead.

**Comment by mlaframboise** [comment]
As an update, this bug still occurs as of build v2.0.7 RC2. It is not possible to log any events on iOS; it still tries to call trackTiming instead of trackEvent when using the buildEvent function.

**Comment by lrgsteven** [comment]
I can confirm the above report on iOS

![timingtesthit](https://cloud.githubusercontent.com/assets/6361385/3305796/63ddef8e-f65a-11e3-8b8b-519afbc7533b.PNG)

The above line was debugged on the device to reveal:

![timing](https://cloud.githubusercontent.com/assets/6361385/3305803/7467f728-f65a-11e3-9eee-3189c06f7af4.PNG)

I even tried another test case of 

var testHit2:Hit = m_googleTracker.buildView( "testaroo" ).create();

which also created a Hit with type of TIMING

**Comment by lrgsteven** [comment]
I tried these same debug tests on Android and the Hits have the correct types.

It seems that the Enum code is not functioning correctly on iOS.

I am using Adobe Air 13 and iOS 7

**Comment by alebianco** [comment]
That's super weird. Are you trying on a device or on the simulator?
I'll do some more tests around this and thanks for the extra details, it will help me a lot.

**Comment by lrgsteven** [comment]
I did the tests on the device and used a remote debugger.

My text editor is Flash Develop (Windows). However, all builds use Ant and Apache Flex http://flex.apache.org/

Thanks for your help. The ANE is much appreciated.

**Comment by alebianco** [comment]
ok. I used to run most of my tests on the simulator on mac and never noticed it.
what about you @mlaframboise ? you are on mac or win?

**Comment by mlaframboise** [comment]
I use the same environments as @lrgsteven (we're co-workers and he offered to help me look into the issue more).

**Comment by alebianco** [comment]
good to know, it might be a packagin problem on windows (i don't think the IDE is to blame)
i'll focus my tests on that, thanks

**Comment by mlaframboise** [comment]
On possible workaround that could be done temporarily is to put the EVENT HitType on the bottom in HitType.as

It looks like iOS is just defaulting to the last HitType (in this case TIMING) regardless of what it should be, so if EVENT is at the bottom, it would make the ANE temporarily usable for the time being until a proper fix is implemented since we're only ever making buildEvent calls.

**Comment by mlaframboise** [comment]
Have you managed to make any progress on this bug?

**Comment by lrgsteven** [comment]
Howdy,
I finally got around to compiling from source and my build did not have the error seen in the binary distribution. Though, I noticed that the user.properties.eg was pointing to iPhoneSimulator instead of iPhoneOS and I can't help but wonder if that is the culprit. Was the binary built with the Simulator SDK instead of the device SDK?
